File: arrays/kth-row-of-pascals-triangle.py
class Solution:
    pascal = [[1], [1, 1]]

    def build_triangle(self, pascal, A):
        while len(pascal) <= A:
            row = [1]
            for i in range(1, len(pascal[-1])):
                row.append(pascal[-1][i-1] + pascal[-1][i])
            row.append(1)
            pascal.append(row)

        return pascal[A]

    def getRow(self, A):
        return self.build_triangle(self.pascal, A)

File: arrays/kth-row-of-pascals-triangle.cpp
vector<int> Solution::getRow(int A) {
    vector<int> v;
    vector<int> w;
    v.push_back(1);
    
    for(int i=1; i<=A; i++){
        w.clear();
        w.push_back(1);
        
        for(int j=0; j<i-1; j++){
            w.push_back(v[j]+v[j+1]);
        }
        
        w.push_back(1);
        v.assign(w.begin(), w.end());
    }
    return v;
}
